spyder/widgets/sourcecode/base.py

```diff
@@ -793,25 +793,19 @@ def go_to_previous_cell(self):
         cursor = self.textCursor()
         cur_pos = prev_pos = cursor.position()
 
-        # Move to the begining of the cell
-        while not self.is_cell_separator(cursor):
-            # Moving to the previous code cell
+        if self.is_cell_separator(cursor):
+            # Move to the previous cell

```

Comment:
@rlaverde I made the change in the comment below this one, it's ok there or should I make the change here?